src/components/ToggleSwitch.js

```diff
@@ -39,12 +40,17 @@ class ToggleSwitch extends React.Component {
     this.props.onToggle(event);
   };
 
-  render () {
+  _handleKeyDown = (event) => {
+    event.preventDefault();
+    if (isEnterOrSpaceKey(event)) this._handleToggle(event);
+  };
+
+  render() {
     const theme = StyleUtils.mergeTheme(this.props.theme);
     const styles = this.styles(theme);
 
     return (
-      <div className='toggle-switch-component' ref={this.props.elementRef} style={styles.component}>
```

Comment:
Would it make sense to change this parent container to a semantic button and then remove the onclick from the individual labels and switch div and instead have it on the parent component?

Reply:
Yeah, it would make more sense to do that. I'll go ahead and change it.

Reply:
If you change it to a semantic button, then you shouldn't have to check for enter or space on a keypress or add a manual tabIndex. You'll get both for free.

Just make sure you're overriding the native button styles cross-browser

Reply:
Just removed the util that check for the space or enter keypress.

Thank you native elements 🙏